### DISCUSSION TITLE
如何使用tailwind 在umi4 中

### DISCUSSION THREAD

**Discussion by chenJJ-88** [umijs/umi#12238] [Q&A]
![image](https://github.com/umijs/umi/assets/118970817/d7a0482e-fc43-4c8d-bcf7-228e8c22b7e6)
好像不能识别这种写法，该怎么修改呢我看自带的是
![image](https://github.com/umijs/umi/assets/118970817/2a46755d-c954-4761-b400-4bec3cb94c7a)
是需要重新下载一个嘛

**Comment by fz6m** [comment]
在 umi 4 里用 tailwind ，目前两种方法：

1. umi 本身就是一个 webpack 项目，看 tailwind 文档按照官方配置方法解决就行，如果要修改 postcss 插件，就配到 [`extraPostCSSPlugins`](https://umijs.org/docs/api/config#extrapostcssplugins) ，然后就可以用了。

2. 执行 `pnpm umi g tailwindcss` （使用 max 时，命令为 `pnpm max g tailwindcss` ） ，这个命令会自动帮你加入一个 umi 内置的 tailwind 插件，并初始化好 tailwind 之类的配置，就可以用了。但这种方式要求你原来项目是干净的，没有 tailwind 的依赖、配置、代码，这样从 0 加入 tailwind 才可以用，否则选择上面第一种就行。

至于你的这个截图手动修改了 `tailwindcss` 到 `@tailwindcss/postcss7-compat` ，这个你要问问别人，或者研究下为什么有人要把 tailwind 偷偷换了，理论上这里不应该是 `@tailwindcss/postcss7-compat` ，而是一个版本号。

**Reply by chenJJ-88** [reply]
好的谢谢大佬
